tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,286 @@
+        f"name={cronjob_name}",
+        "namespace=default",
+        "--out json",
+        "--log-level quiet",
+    ]
+    result = utils.run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    assert json.loads(result.stdout) == True, "Failed to activate the CronJob"
+
+
+@when(
+    parsers.parse("we wait for '{seconds}' seconds"),
+)
+def wait_seconds(seconds):
+    time.sleep(int(seconds))
+
+
+@when(
+    parsers.parse("we get the '{cronjob_name}' jobs"),
+)
+def call_metalk8s_kubernetes_cronjob_get_jobs(host, ssh_config, context, cronjob_name):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.get_jobs",
+        f"cronjob_name={cronjob_name}",
+        "namespace=default",
+        "--out json",
+        "--log-level quiet",
+    ]
+
+    result = utils.run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    context[f"{cronjob_name}_jobs"] = json.loads(result.stdout)
+
+
+@when(
+    parsers.parse(
+        "we stop the '{cronjob_name}' CronJob and delete all its Jobs marked with '{suspend_mark}'"
+    ),
+)
+def call_metalk8s_kubernetes_cronjob_stop_jobs(
+    host, ssh_config, context, cronjob_name, suspend_mark
+):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.stop_jobs",
+        f"cronjob_name={cronjob_name}",
+        "namespace=default",
+        f"mark={suspend_mark}",
+        "wait=true",
+        "--out json",
+        "--log-level quiet",
+    ]
+
+    result = utils.run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    context[f"deleted_jobs"] = json.loads(result.stdout)
+
+
+# }}}
+# Then {{{
+
+
+@then(
+    parsers.cfparse(
+        "the '{cronjob_name}' CronJob should{negated:Negation?} be "
+        "in the list of '{state}' CronJobs",
+        extra_types={"Negation": negation},
+    ),
+)
+def confirm_cronjob_presence(context, cronjob_name, negated, state):
+    cronjobs = [
+        c for c in context[f"cronjobs_{state}"] if c["metadata"]["name"] == cronjob_name
+    ]
+    if negated:
+        assert not cronjobs, "CronJob is in the list of suspended CronJobs"
+    else:
+        assert cronjobs, "CronJob is not in the list of suspended CronJobs"
+        context[f"cronjob_{cronjob_name}"] = cronjobs[0]
+
+
+@then(
+    parsers.cfparse(
+        "the '{cronjob_name}' CronJob should{negated:Negation?} be "
+        "marked with '{suspend_mark}'",
+        extra_types={"Negation": negation},
+    ),
+)
+def get_suspend_mark(context, cronjob_name, negated, suspend_mark):
+    cronjob = context[f"cronjob_{cronjob_name}"]
+    suspend_annotation = (
+        cronjob.get("metadata", {})
+        .get("annotations", {})
+        .get("metalk8s.scality.com/suspend_mark")
+    )
+    if negated:
+        assert (
+            not suspend_annotation or suspend_annotation != suspend_mark
+        ), "CronJob has a suspend_mark"
```

Comment:
To me it should either be "marked with ...." or "not marked" but there is no case where we would use "not marked with ..." no ?